lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -512,8 +514,9 @@ def _on_opensearch_data_storage_detaching(self, _: StorageDetachingEvent):  # no
                 else:
                     raise OpenSearchHAError(ClusterHealthUnknown)
         finally:
-            # release lock
-            self.node_lock.release()
+            if self.app.planned_units() > 0 and (self.opensearch.is_node_up() or self.alt_hosts):
+                # release lock
+                self.node_lock.release()
```

Comment:
I think this might not be consistent here, because in line 476-480 you're not getting a lock if `planned_units()` == 1, but here you want to release it.

Reply:
Same comment as above. We may want to extend this logic to have a `online_nodes > 1 and self.app.planned_units() > 1`

Reply:
Check my previous answer to your comment at `acquired` call.